.circleci/config.yml

```diff
@@ -0,0 +1,216 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - save_cache:
+          key: v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+          paths:
+            - "downloads"
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-
+
+      - run:
+          name: Install pip packages
+          command: |
+
+            source venv/bin/activate
```

Comment:
I think it will be clearer if we created the venv here too, instead of doing it on the "circle-ci-configuration.sh", would like to hear your thoughts on this.

Reply:
If the virtualenv is created after the previous restore cache step, it resets the virtualenv; which is why I moved it to the `ci-configuration.sh`. But I get your point. Let me see if this can changed so that the virtualenv is only created if the previous restore cache step did not find a cache.